internal/pkg/cli/svc_deploy.go

```diff
@@ -461,19 +481,42 @@ func (o *deploySvcOpts) deploySvc(addonsURL string) error {
 	return nil
 }
 
-func (o *deploySvcOpts) validateAppVersion(svc *manifest.LoadBalancedWebService) error {
-	var appVersion string
+func validateAlias(alias string, app *config.Application, envName string, appVersionGetter versionGetter) error {
+	if alias == "" {
+		return nil
+	}
+	if err := validateAppVersion(alias, app, appVersionGetter); err != nil {
```

Comment:
A random thought: I feel like we can `validateAppVersion` after matching regex. It's possible that the user wants to use a particular alias that is not supported by Copilot. In that case they can't be using Copilot for that alias, hence not needing to upgrade the app..?

Reply:
I think either should be fine! I agree if they want to enable alias and it is not valid (either because of the app version or the regex matching fail), the total amount of time consumed by copilot would be slightly different.